metricproviders/datadog/datadog.go

```diff
@@ -72,16 +95,34 @@ func (p *Provider) Run(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric) v1alph
 		StartedAt: &startTime,
 	}
 
-	endpoint := "https://api.datadoghq.com/api/v1/query"
+	endpoint := "https://api.datadoghq.com"
 	if p.config.Address != "" {
-		endpoint = p.config.Address + "/api/v1/query"
+		endpoint = p.config.Address
 	}
 
+	// Check if the URL is valid first before adding the endpoint
 	url, err := url.Parse(endpoint)
 	if err != nil {
 		return metricutil.MarkMeasurementError(measurement, err)
 	}
 
+	apiVersion := "v1"
+	if metric.Provider.Datadog.ApiVersion != "" {
+		apiVersion = metric.Provider.Datadog.ApiVersion
+	}
+
+	if apiVersion == "v1" {
+		p.logCtx.Warn("Datadog will soon deprecate their API v1. Please consider switching to v2 soon.")
+	}
+
+	route := "/api/v1/query"
+	if apiVersion == "v2" {
+		route = "/api/v2/query/timeseries"
+	}
+
+	// Add endpoint after getting the API version
+	url, _ = url.Parse(endpoint + route)
```

Comment:
Can we handle this error

Reply:
Sorry, can you elaborate further? Not sure what you mean by this

Reply:
you are sending the err to _ I would like to handle it, even though both endpoint and route variables are static and controlled in code to future proof say they become user configurable in the future etc it would be nice just to have the error handheld incase Parse fails for any reason.

Reply:
Commit: https://github.com/argoproj/argo-rollouts/pull/2592/commits/2799dc91318264296f553a85d818781ce6b4ba0b